File: ultimate/AuthorizationMiddlewareHandler.java
package ultimate;

import chainofresponsibility.AbstractMiddlewareHandler;

public class AuthorizationMiddlewareHandler extends AbstractMiddlewareHandler {

    @Override
    public MiddlewareResponse handle(HttpRequest request) {
        if (request.getUser().isAuthorized()) {
            super.handle(request);
        }
        return new MiddlewareResponse("Request has failed to be authorized", false);
    }
}

File: typescript/patterns/ultimate-design-pattern-challenge/authorization-middleware-handler.ts
class AuthorizationMiddlewareHandler extends AbstractMiddlewareHandler {
  override handle(request: HttpRequest): MiddlewareResponse {
    if (!request.getUser().getIsAuthorized()) {
      return new MiddlewareResponse(
        "Request has failed to be authorized",
        false,
      );
    }

    return super.handle(request);
  }
}
